server/data-migrations/1742369617678-EnforceNewBasicPlanLimits.ts

```diff
@@ -0,0 +1,174 @@
+import { Brackets, EntityManager, MigrationInterface, QueryRunner } from 'typeorm';
+import { AppModule } from '@modules/app/module';
+import { NestFactory } from '@nestjs/core';
+import { LicenseCountsService } from '@ee/licensing/services/count.service';
+import { dbTransactionWrap } from '@helpers/database.helper';
+import { OrganizationUser } from '@entities/organization_user.entity';
+import { USER_STATUS, USER_TYPE, WORKSPACE_USER_STATUS } from '@modules/users/constants/lifecycle';
+import { User } from '@entities/user.entity';
+import { USER_ROLE } from '@modules/group-permissions/constants';
+import { LicenseInitService } from '@modules/licensing/interfaces/IService';
+export class EnforceNewBasicPlanLimits1742369617678 implements MigrationInterface {
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    const manager = queryRunner.manager;
+    const nestApp = await NestFactory.createApplicationContext(await AppModule.register({ IS_GET_CONTEXT: true }));
+    const licenseInitService = nestApp.get(LicenseInitService);
+
+    try {
+      const { isValid } = await licenseInitService.initForMigration(manager);
+      console.log('isValid', isValid);
+      if (!isValid) {
+        await dbTransactionWrap(async (manager: EntityManager) => {
+          const licenseCountsService = nestApp.get(LicenseCountsService);
+
+          const statusList = [WORKSPACE_USER_STATUS.INVITED, WORKSPACE_USER_STATUS.ACTIVE];
+
+          const usersWithEditPermission = (
```

Comment:
Use native postgres query in migration files
minimise use of createQueryBuilder